sources/sns/topic.go

```diff
@@ -0,0 +1,87 @@
+package sns
+
+import (
+	"context"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/sns"
+	"github.com/overmindtech/aws-source/sources"
+	"github.com/overmindtech/sdp-go"
+)
+
+type topicClient interface {
+	GetTopicAttributes(ctx context.Context, params *sns.GetTopicAttributesInput, optFns ...func(*sns.Options)) (*sns.GetTopicAttributesOutput, error)
+	ListTopics(context.Context, *sns.ListTopicsInput, ...func(*sns.Options)) (*sns.ListTopicsOutput, error)
+	ListTagsForResource(context.Context, *sns.ListTagsForResourceInput, ...func(*sns.Options)) (*sns.ListTagsForResourceOutput, error)
+}
+
+func getTopicFunc(ctx context.Context, client topicClient, scope string, input *sns.GetTopicAttributesInput) (*sdp.Item, error) {
+	output, err := client.GetTopicAttributes(ctx, input)
+	if err != nil {
+		return nil, err
+	}
+
+	if output.Attributes == nil {
+		return nil, &sdp.QueryError{
+			ErrorType:   sdp.QueryError_NOTFOUND,
+			ErrorString: "get topic attributes response was nil",
+		}
+	}
+
+	attributes, err := sources.ToAttributesCase(output.Attributes)
+	if err != nil {
+		return nil, err
+	}
+
+	resourceTags, err := tagsByResourceARN(ctx, client, *input.TopicArn)
+	if err != nil {
+		return nil, err
+	}
+
+	item := &sdp.Item{
+		Type:            "sns-topic",
+		UniqueAttribute: "topicArn",
+		Attributes:      attributes,
+		Scope:           scope,
+		Tags:            tagsToMap(resourceTags),
+	}
```

Comment:
We could also link these via the `KmsMasterKeyId` to a KMS key. The source doesn't exist yet but it should be called `kms-key` as per the naming convention so we should be able to do a GET using the ID

Reply:
Great catch, thx!

https://github.com/overmindtech/aws-source/pull/415/commits/b9a458cfa0da9a710053b306833a8055bbffe1b4